src/test/components/MenuButtons.test.js

```diff
@@ -47,10 +50,21 @@ describe('app/MenuButtons', function() {
       rejectUpload = reject;
     });
 
+    promise.catch(() => {
+      // Node complains if we don't handle a promise/catch, and this one may reject
+      // before it's properly handled. Catch it here so that Node doesn't complain.
+    });
```

Comment:
This was already done like this in publish.test.js. This happens if in one test we create the promise here, we reject it when aborting, but actually never retrieved running `startUpload`, because the request is canceled before reaching that point in `attemptToUpload`.

Reply:
This make me a bit nervous that a regression could slide through. I'm not really how to properly wait for things here though without digging in.

Reply:
I tried doing it differently as we discussed on riot  but this felt overengineered. Looking at the code I think we won't miss a promise rejection anyway because we "await" this promise in the app code, so any rejection will be handled there, or not handled. So I just added 2 comment lines to explain this :)